File: android/app/src/main/java/com/transporthq/app/data/models/Notification.kt
package com.transporthq.app.data.models

import com.google.gson.annotations.SerializedName

data class AppNotification(
    @SerializedName("_id") val id: String = "",
    val title: String = "",
    val message: String = "",
    val type: String = "",
    val read: Boolean = false,
    val userId: String = "",
    val data: NotificationData? = null,
    val createdAt: String? = null
)

data class NotificationData(
    val rideRequestId: String? = null,
    val tripId: String? = null,
    val driverId: String? = null
)

data class NotificationsResponse(
    val notifications: List<AppNotification> = emptyList()
)

data class UnreadCountResponse(
    val count: Int = 0
)

File: ios/TransportHQ/Models/Notification.swift
import Foundation

struct AppNotification: Codable, Identifiable {
    let id: String
    let type: String?
    let title: String?
    let message: String?
    let data: NotificationData?
    let read: Bool?
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case type, title, message, data, read, createdAt
    }
}

struct NotificationData: Codable {
    let rideRequestId: String?
    let tripId: String?
    let driverId: String?
}

struct UnreadCountResponse: Codable {
    let count: Int
}

struct DriverLocation: Identifiable {
    let id: String
    var lat: Double
    var lng: Double
    var speed: Double
    var heading: Double
    var driverName: String
    var updatedAt: Date

    init(id: String, lat: Double, lng: Double, speed: Double = 0, heading: Double = 0, driverName: String = "", updatedAt: Date = Date()) {
        self.id = id
        self.lat = lat
        self.lng = lng
        self.speed = speed
        self.heading = heading
        self.driverName = driverName
        self.updatedAt = updatedAt
    }
}
